projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -3400,11 +3400,16 @@ public function untrash_feedback_status_handler( $current_status, $post_id, $pre
 	 * Tracks when a feedback post status changes to 'spam' and stores the timestamp.
 	 * This allows us to accurately determine when spam was marked, independent of other post updates.
 	 *
-	 * @param string  $new_status The new post status.
-	 * @param string  $old_status The old post status.
-	 * @param WP_Post $post       The post object.
+	 * @param string       $new_status The new post status.
+	 * @param string       $old_status The old post status.
+	 * @param WP_Post|null $post       The post object, when available.
 	 */
-	public function track_spam_status_change( $new_status, $old_status, WP_Post $post ) {
+	public function track_spam_status_change( $new_status, $old_status, ?WP_Post $post = null ) {
+		if ( ! $post instanceof WP_Post ) {
```

Comment:
Isn't this gonna trigger an error if `$post` is null? Double check just in case with `! $post || ! $post instanceof WP_Post`?